### PR TITLE
Player Plugin Loading

### PR DESCRIPTION
**Pull request by lkiesow**
This patch switches to the OSGi annotations for the Theodul bundles,
fixing the problem that there were two plugin managers deployed in
Opencast by accident which caused all plugin to be loaded twice:

    2021-07-29T02:01:27,286 | INFO  | (EngagePluginManagerImpl:156) - Installed Theodul plugin EngagePluginVideoVideoJS (static: 1/static REST: no)
    2021-07-29T02:01:27,291 | INFO  | (EngagePluginManagerImpl:156) - Installed Theodul plugin EngagePluginVideoVideoJS (static: 2/static REST: no)
    2021-07-29T02:01:27,292 | INFO  | (EngagePluginVideoVideojs:38) - Activated engage plugin: Videojs.

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

File: modules/engage-theodul-plugin-tab-shortcuts/src/main/java/org/opencastproject/engage/theodul/plugin/tab/shortcuts/EngagePluginTabShortcuts.java [hunk 1]
```diff
@@ -22,19 +22,40 @@
...
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.ws.rs.Path;
 
 @Path("/")
+@Component(
+    property = {
+        "opencast.engage.plugin.name=EngagePluginTabShortcuts",
+        "opencast.engage.plugin.shortcuts=Simple implementation of a tab with the shortcuts",
```
Comment (mtneug) [inline modules/engage-theodul-plugin-tab-shortcuts/src/main/java/org/opencastproject/engage/theodul/plugin/tab/shortcuts/EngagePluginTabShortcuts.java]: This should be `opencast.engage.plugin.description` (wrong in the original XML).

File: modules/engage-theodul-plugin-tab-downloads/src/main/java/org/opencastproject/engage/theodul/plugin/tab/downloads/EngagePluginTabDownloads.java [hunk 2]
```diff
@@ -53,8 +53,8 @@ protected void activate(ComponentContext cc) {
   }
 
   @Reference
-  protected void setPluginManager(EngagePluginManager pm) {
-    super.setPluginManager(pm);
+  protected void setPluginManager(final EngagePluginManager pluginManager) {
```
Comment (mtneug) [inline modules/engage-theodul-plugin-tab-downloads/src/main/java/org/opencastproject/engage/theodul/plugin/tab/downloads/EngagePluginTabDownloads.java:57]: * Why add the `final` only here?
* Why not adding `@Override` only here?

File: modules/engage-theodul-plugin-custom-notifications/src/main/java/org/opencastproject/engage/theodul/plugin/custom/notifications/EngagePluginCustomNotifications.java [hunk 3]
```diff
@@ -22,19 +22,39 @@
...
-  protected void activate(ComponentContext cc) {
-    log.info("Activated engage plugin: Notifications.");
+  @Activate
+  protected void activate() {
+    logger.info("Activated engage plugin: Notifications.");
+  }
+
+  @Reference
+  @Override
+  protected void setPluginManager(final EngagePluginManager pluginManager) {
```
Comment (mtneug) [inline modules/engage-theodul-plugin-custom-notifications/src/main/java/org/opencastproject/engage/theodul/plugin/custom/notifications/EngagePluginCustomNotifications.java:57]: Why add `final` only here?

**Review by mtneug — CHANGES_REQUESTED** [review]
Only minor comments on first round.

File: modules/engage-theodul-plugin-tab-shortcuts/src/main/java/org/opencastproject/engage/theodul/plugin/tab/shortcuts/EngagePluginTabShortcuts.java (hunk 1 above)
Comment (lkiesow) [reply modules/engage-theodul-plugin-tab-shortcuts/src/main/java/org/opencastproject/engage/theodul/plugin/tab/shortcuts/EngagePluginTabShortcuts.java]: fixed

File: modules/engage-theodul-plugin-tab-downloads/src/main/java/org/opencastproject/engage/theodul/plugin/tab/downloads/EngagePluginTabDownloads.java (hunk 2 above)
Comment (lkiesow) [reply modules/engage-theodul-plugin-tab-downloads/src/main/java/org/opencastproject/engage/theodul/plugin/tab/downloads/EngagePluginTabDownloads.java:57]: fixed

File: modules/engage-theodul-plugin-custom-notifications/src/main/java/org/opencastproject/engage/theodul/plugin/custom/notifications/EngagePluginCustomNotifications.java (hunk 3 above)
Comment (lkiesow) [reply modules/engage-theodul-plugin-custom-notifications/src/main/java/org/opencastproject/engage/theodul/plugin/custom/notifications/EngagePluginCustomNotifications.java:57]: fixed

**Review by mtneug — APPROVED** [review]
Works as intended.

**Pull request merged**